nym-vpn-desktop/src-tauri/src/commands/credential.rs

```diff
@@ -0,0 +1,36 @@
+use nym_vpn_lib::credentials::import_credential_base58;
+use tauri::State;
+use tracing::{debug, error, instrument};
+
+use crate::{
+    db::{Db, Key},
+    error::{CmdError, CmdErrorSource},
+    fs::path::BACKEND_DATA_PATH,
+};
+
+#[instrument(skip_all)]
+#[tauri::command]
+pub async fn add_credential(credential: String, db: State<'_, Db>) -> Result<(), CmdError> {
+    debug!("add_credential");
+
+    debug!(
+        "using path for credential: {}",
+        BACKEND_DATA_PATH.to_string_lossy()
+    );
+    import_credential_base58(&credential, BACKEND_DATA_PATH.clone())
+        .await
+        .map_err(|e| {
+            error!("failed to import credential: {:?}", e);
+            // TODO improve error handling, distinguish between
+            // errors linked to user responsibility or internal errors
+            CmdError::new(
+                CmdErrorSource::InternalError,
+                "failed to import credential".to_string(),
+            )
+        })?;
+
+    db.insert(Key::ImportedCredential, true)
```

Comment:
yes noted

anyway this k/v was "an attempt" to record when user imported a valid cred, but it's suboptimal since we cant keep it in sync with real cred existence (stored in FS)
so better to remove it completely